File: 001. Two Sum/Solution.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        m_map = {}
        for i, v in enumerate(nums):
            if target - v in m_map:
                return m_map[target-v], i
            else:
                m_map[v] = i
        return []

File: 001. Two Sum/Solution.java
public class Solution {
    public int[] twoSum(int[] numbers, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int i=0; i < numbers.length; i ++){
            map.put(numbers[i], i);
        }
        int[] result = new int[2];
        for(int i = 0; i < numbers.length; i ++){
            int waitCal = target-numbers[i];
            if(map.get(waitCal) != null && map.get(waitCal) > i){
                result[0] = i;
                result[1] = map.get(waitCal);
                break;
            }
        }   
    return result;
    }
}